### PR TITLE
readable blob discretisation

### PR DESCRIPTION
**Pull request by gregordecristoforo**
Implementation of readable blob discretisation

### CODE REVIEW

File: blobmodel/blobs.py [hunk 1]
```diff
@@ -20,34 +20,46 @@ class Blob():
...
         elif(self.blob_shape == 'exp'):
-            return self.amplitude * np.exp(-(t-self.t_init)/self.t_drain)\
-                * np.exp(x - self.pos_x - self.v_x*(t-self.t_init))\
-                * np.exp(-((y-self.pos_y - self.v_y*(t-self.t_init))**2/(2*self.width_y**2)))\
-                * np.heaviside(t-self.t_init, 1)*np.heaviside(-1.*(x - self.pos_x - self.v_x*(t-self.t_init)), 1)
+            return self.amplitude\
+                * self.__drain(t)\
+                * self.__x_shape_exp(x, t)\
+                * self.__y_shape_gauss(y, t)\
+                * self.__blob_arrival(t)
```
Comment (Sosnowsky) [inline blobmodel/blobs.py:40]: I think there should be no if-else statement here. Instead only something like 
            return self.amplitude\
                * self.__drain(t)\
                * self.__x_shape(x, t)\
                * self.__y_shape(y, t)\
                * self.__blob_arrival(t)
              
And then the x_shape and y_shape functions will have the if-else statement. This way you reduce code duplication

**Review by Sosnowsky — CHANGES_REQUESTED** [review]

File: blobmodel/blobs.py (hunk 1 above)
Comment (gregordecristoforo) [reply blobmodel/blobs.py]: I changed the code accordingly, what do you think now?

**Comment by gregordecristoforo** [comment]
The relevant code is now     

def discretize_blob(self, x, y, t):
        '''
        Discretize blob on grid
        The following blob shapes are implemented:
                gauss: 2D gaussian function
                exp: one sided exponential in x and gaussian in y
        '''
        return self.amplitude\
            * self.__drain(t)\
            * self.__x_shape(x, t, self.blob_shape)\
            * self.__y_shape(y, t)\
            * self.__blob_arrival(t)
            * 
    def __x_shape(self, x, t, blob_shape):
        if blob_shape == 'gauss':
            return np.exp(-((x - self.__get_x_blob_pos(t))**2/(2*self.width_x**2)))
        elif blob_shape == 'exp':
            return np.exp(x - self.__get_x_blob_pos(t))\
                * np.heaviside(-1.*(x - self.__get_x_blob_pos(t)), 1)
        else:
            raise NotImplementedError(
                self.__class__.__name__ + '.blob shape not implemented')

**Review by Sosnowsky — CHANGES_REQUESTED** [review]

File: blobmodel/blobs.py [hunk 2]
```diff
@@ -20,34 +20,39 @@ class Blob():
...
-                * np.exp(-((y-self.pos_y - self.v_y*(t-self.t_init))**2/(2*self.width_y**2)))\
-                * np.heaviside(t-self.t_init, 1)
-        elif(self.blob_shape == 'exp'):
-            return self.amplitude * np.exp(-(t-self.t_init)/self.t_drain)\
-                * np.exp(x - self.pos_x - self.v_x*(t-self.t_init))\
-                * np.exp(-((y-self.pos_y - self.v_y*(t-self.t_init))**2/(2*self.width_y**2)))\
-                * np.heaviside(t-self.t_init, 1)*np.heaviside(-1.*(x - self.pos_x - self.v_x*(t-self.t_init)), 1)
+        return self.amplitude\
+            * self.__drain(t)\
+            * self.__x_shape(x, t, self.blob_shape)\
```
[the remark this answers is not in the document]
Comment (gregordecristoforo) [reply blobmodel/blobs.py]: thx! argument is now removed

**Review by Sosnowsky — APPROVED** [review]

**Pull request merged**